src/main/java/com/puppycrawl/tools/checkstyle/site/ViolationMessagesMacro.java

```diff
@@ -0,0 +1,274 @@
+
+import com.puppycrawl.tools.checkstyle.ModuleFactory;
+import com.puppycrawl.tools.checkstyle.PackageNamesLoader;
+import com.puppycrawl.tools.checkstyle.PackageObjectFactory;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.checks.coding.AbstractSuperCheck;
+import com.puppycrawl.tools.checkstyle.checks.naming.AbstractAccessControlNameCheck;
+import com.puppycrawl.tools.checkstyle.checks.naming.AbstractNameCheck;
+import com.puppycrawl.tools.checkstyle.checks.regexp.RegexpMultilineCheck;
+import com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineCheck;
+import com.puppycrawl.tools.checkstyle.checks.regexp.RegexpSinglelineJavaCheck;
+import com.puppycrawl.tools.checkstyle.checks.whitespace.AbstractParenPadCheck;
+
+/**
+ * A macro that inserts an unordered list of the violation messages.
+ */
+@Component(role = Macro.class, hint = "violation-messages")
+public class ViolationMessagesMacro extends AbstractMacro {
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        final String moduleName = (String) request.getParameter("moduleName");
+        final ModuleFactory moduleFactory = getModuleFactory();
+        final Object instance = getModuleInstance(moduleName, moduleFactory);
+        final Class<?> clss = instance.getClass();
+        final Set<Field> messageKeyFields = getCheckMessageKeyFields(clss);
+        final Set<String> messageKeys = getMessageKeys(messageKeyFields, instance);
+        createListOfMessages(sink, clss, messageKeys);
+    }
+
+    /**
+     * Returns the default PackageObjectFactory with the default package names.
+     *
+     * @return the default PackageObjectFactory.
+     * @throws MacroExecutionException if the PackageObjectFactory could not be created.
+     */
+    private static ModuleFactory getModuleFactory() throws MacroExecutionException {
+        try {
+            final ClassLoader cl = ViolationMessagesMacro.class.getClassLoader();
+            final Set<String> packageNames = PackageNamesLoader.getPackageNames(cl);
+            return new PackageObjectFactory(packageNames, cl);
+        }
+        catch (CheckstyleException ex) {
+            throw new MacroExecutionException("Couldn't load checkstyle modules", ex);
+        }
+    }
+
+    /**
+     * Returns the checkstyle messages for the given class.
+     *
+     * @param clss the class.
+     * @return the checkstyle messages.
+     * @throws MacroExecutionException if the messages could not be retrieved.
+     */
+    private static Set<Field> getCheckMessageKeyFields(Class<?> clss)
+            throws MacroExecutionException {
+        final Set<Field> checkstyleMessages = new HashSet<>();
+
+        final Field[] fields = clss.getDeclaredFields();
+
+        for (Field field : fields) {
+            if (field.getName().startsWith("MSG_")) {
+                checkstyleMessages.add(field);
+            }
+        }
```

Comment:
```
Set s = Arrays.stream(fields)
  .filter(f -> ...)
  .Collect(Collections.toSet);
```

Reply:
I changed the method(`getCheckMessageKeys`) to get the `String` of the message keys. Doing this lambda would require me to do 
```java
    final List<String> checkstyleMessageKeys = Arrays.stream(fields)
            .filter(field -> field.getName().startsWith("MSG_"))
            .map(field -> {
                try {
                    return getFieldValue(field, checkInstance);
                }
                catch (IllegalAccessException exception) {
                    throw new RuntimeException(exception);
                }
            })
            .collect(Collectors.toList());
```
which doesn't look good in my opinion.

Reply:
Agreed.